frontend/src/views/UsersList.js

```diff
@@ -0,0 +1,227 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import apiConfig from 'base/apiConfig';
+import { trackPromise } from 'react-promise-tracker';
+import styles from './_users-list-content.scss';
+import HeaderAreaLayout from 'base/components/layout/HeaderAreaLayout';
+import HeaderContentStatic from 'base/components/header-views/header-content-static/HeaderContentStatic';
+import Paginator from 'base/components/layout/Paginator';
+import ListSearch from 'base/components/inputs/ListSearch';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faCheck, faAngleDoubleUp, faAngleDoubleDown } from '@fortawesome/free-solid-svg-icons';
+
+import classNames from 'classnames/bind';
+let cx = classNames.bind(styles);
+
+
+class UsersList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      usersList: [],
+      perPage: 20,
+      count: 0,
+      pages: 0,
+      currentPage: 1,
+      searchQuery: '',
+      ordering: 'profile__name',
+    };
+
+    this.getUsers = this.getUsers.bind(this);
+    this.setPerPage = this.setPerPage.bind(this);
+    this.setSearchQuery = this.setSearchQuery.bind(this);
+    this.setOrdering = this.setOrdering.bind(this);
+  }
+
+  getUsers(page = 1) {
+    const offset = (page-1) * this.state.perPage;
+    const requestUrl = apiConfig.learnersGeneral + '?search=' + this.state.searchQuery + '&ordering=' + this.state.ordering + '&limit=' + this.state.perPage + '&offset=' + offset;
```

Comment:
@grozdanowski Perhaps add a function to this class to do the url construction or at least break this into multiple lines for readability. I'm starting to look at how we want to be able to add tests to Figures front end, and url constructors like this look like good candidates for specific tests to make sure the front and and back end are consistent

Reply:
@johnbaldwin thanks for the suggestion. I made the change.